File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        visited=set()
        count=0
        def helper(idx):
            visited.add(idx)
            for i in range(len(isConnected[idx])):
                if i==idx or isConnected[idx][i]==0 or i in visited:
                    continue;

                helper(i)
      
        for i in range(len(isConnected)):
            if i in visited:
                continue;
            count+=1
            helper(i)

        return count

File: 0547-number-of-provinces/0547-number-of-provinces.js
/**
 * @param {number[][]} isConnected
 * @return {number}
 */
var findCircleNum = function(isConnected) {
    let visited=new Set(),count=0
    function helper(idx){
        visited.add(idx)
        for(let i=0;i<isConnected[idx].length;i++){
            if(i===idx || isConnected[idx][i]===0 || visited.has(i)) continue;
            
            helper(i)
        }
    }
    for(let i=0;i<isConnected.length;i++){
        if(visited.has(i)) continue;
        count++
        helper(i)
        
    }
    return count
};